### ISSUE TITLE
`CelestialCondition` within `AltitudeEvents` class.

### ISSUE THREAD

**Issue by Tronald** [Tronald/CoordinateSharp#161]
User Request:

If a `CelestialCondition` could return within the `AltitudeEvents` class, it would allow for users to determine failure conditions as to why a rise or set has not occurred within a specified altitude.

Current `AltitudeEvents` class should be expanded to allow for this property. No effect on performance is expected as the logic is already running, the property simply needs to be assigned.

**Comment by Tronald** [comment]
`Condition` property of type `CelestialStatus` has been added to the `AltitudeEvents` class.

This will allow users to determine why one or both `Rising` and `Setting` properties have return null. 

For example:

```C#
AltitudeEvents altE = Celestial.Get_Time_at_Solar_Altitude(47, -122, new DateTime(2020,9,30), 54);
Console.WriteLine(altE.Condition); //Returns DownAllDay as the sun never reaches 54 degrees of altitude.
```

This will be available with v2.8.1.1. We are currently behind on release due to work loads at the moment. Keeping open until the next release is published.

**Comment by Tronald** [comment]
Release with v2.8.1.1

Example:
```C#
Coordinate c = new Coordinate(47.40615, -122.24517, new DateTime(2020, 8, 11));

//Determine times the sun passes through 57.8 degrees.
var t = Celestial.Get_Time_at_Solar_Altitude(c, 57.8);
t.Rising; //Null
t.Setting; //Null;

//Should show DownAllDay as the sun never reaches a height of 57.8 degree at the specified location and date.
t.Condition; //DownAllDay
```
